File: 프로그래머스/2/84512. 모음 사전/모음 사전.js
let idx = 0;
const vowels = ['A', 'E', 'I', 'O', 'U']
const res = {}

function solution(word) {
    dfs('', 0)
    return res[word]
}

function dfs(word, length){
    if(length>5)    return;
    res[word] = idx++;
    
    for(let i=0; i<vowels.length; i++){
        dfs(word+vowels[i], length+1)
    }
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
from itertools import product

def solution(word):
    words = []
    vowels = ['A', 'E', 'I', 'O', 'U']
    
    for length in range(1, 6):
        for p in product(vowels, repeat=length):
            words.append(''.join(p))
    
    words.sort()
    return words.index(word) + 1